### DISCUSSION TITLE
[Deployment] Assertion error: factory deps non zero, data: "0x"

### DISCUSSION THREAD

**Discussion by wwwMalcolm** [zkSync-Community-Hub/zksync-developers#787] [Deployments] [answered]
### Environment

Testnet

### zkSolc Version

zksolc-1.5.7

### zksync-ethers Version

solc 0.8.28

### Minimal Reproducible Code or Repo Link

forge create src/Counter.sol:Counter \
    --account myKeystore \
    --rpc-url https://sepolia.era.zksync.dev \
    --chain 300 \
    --zksync

Ran the command above but get this error below:
server returned an error response: error code 3: virtual machine entered unexpected state. please contact developers and provide transaction details that caused this error. Error description: Assertion error: factory deps non zero, data: "0x"

### Does this work on other EVMs? (If yes, please list at least 1 of them)

No

**Comment by alhenaai** [comment]
It appears you are encountering a specific error related to the deployment of a contract on the ZKsync Testnet using the `forge create` command. The error message "Assertion error: factory deps non zero, data: '0x'" suggests an issue with the virtual machine's state during the transaction execution.

This type of error can be complex and might require detailed investigation by the development team. Here are a few steps you can take:

1. **Verify the Contract and Environment**: Ensure that your contract code is correct and compatible with the ZKsync environment. Since it does not work on other EVMs, there might be an issue with how the contract interacts with the ZKsync-specific features or limitations.

2. **Check the Network and Parameters**: Double-check the network parameters and the RPC URL you are using. Make sure that they are correct for the ZKsync Testnet environment.

3. **Contact Support**: Since the error message suggests contacting developers, it would be beneficial to report this issue directly to the ZKsync development team. You can create an issue in the [ZKsync developers discussions on Github](https://github.com/zkSync-Community-Hub/zksync-developers/discussions) where you can provide the transaction details and the error encountered for further assistance.

By reporting this, the ZKsync team can provide more specific guidance or patches if there's a bug affecting your deployment process.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by uF4No** [comment]
Can you confirm you're using [foundry-zksync](https://docs.zksync.io/build/tooling/foundry/overview)?

If so, please run `forge --version` and share which version are you running to try to replicate the issue.

**Reply by wwwMalcolm** [reply]
forge 0.0.2 (f75d7bf 2024-11-22T00:26:55.917059070Z)

**Reply by wwwMalcolm** [reply]
I tried doing another foundry init following this https://docs.zksync.io/build/tooling/foundry/getting-started

And now I get this error when running foundry create instead.
`Error: server returned an error response: error code 3: Failed to serialize transaction: toAddressIsNull`

**Reply by uF4No** [reply, marked as answer]
Looks like the latest nightly version has a bug  ([reported here](https://github.com/matter-labs/foundry-zksync/issues/738)). Please install the previous version with `foundryup-zksync --version nightly-420660c5243e06af1f12febb1765a9abc9c77461`. Once installed make sure `forge --version` returns `forge 0.0.2 (420660c 2024-11-21T00:41:59.430174000Z)`

You should be able to deploy the contract with no issues. I just tried on my end:

```
forge create src/Counter.sol:Counter --account myKeystore --rpc-url https://sepolia.era.zksync.dev --chain 300 --zksync 

[⠃] Using zksolc-1.5.7
[⠊] Compiling (zksync)
No files changed, compilation skipped
Enter keystore password: **********************
Deployer: 0x466ff3c5C76445823b49dF047d72663B8eAe9272
Deployed to: 0x3BfF4A55De1c00Dc04132FD49F0A78993DF4f83c
Transaction hash: 0xf1aca008991d180ea008c644d316ad9d29344257c53905c359ef2ac061119529
```
